src/main/java/ChatClient.java

```diff
@@ -0,0 +1,54 @@
+import java.io.*;
+import java.net.Socket;
+
+public class ChatClient {
+    private Socket socket;
+    private UserIO io;
+
+    public ChatClient(UserIO io, Socket socket) {
+        this.io = io;
+        this.socket = socket;
+    }
+
+    public void writeToServer() {
+        io.showInitialMessage(socket.getPort());
+        String message = io.getInput();
+        try {
+            PrintWriter printWriter = createPrintWriter();
+            writeMessageTillQuit(message, printWriter);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        io.showExitMessage();
+        closeSocket();
+    }
+
+    private void writeMessageTillQuit(String name, PrintWriter printWriter) {
+        while (!name.contains("quit")) {
+            printWriter.println(name);
+            printWriter.flush();
+            try {
+                InputStream inputFromClient = socket.getInputStream();
```

Comment:
This is quite a lot of boilerplate code - could you pull out the construction of the reader into a method?